keanu-project/src/main/java/io/improbable/keanu/model/linear/LinearRegression.java

```diff
@@ -60,7 +62,7 @@ private BayesianNetwork buildModel(DoubleTensor x, double priorOnMu, double prio
     }
 
     private static double[] fillPriorOnWeights(DoubleTensor x, double priorOnSigma) {
-        double[] priorWeights = new double[x.getShape()[0]];
+        double[] priorWeights = new double[Ints.saturatedCast(x.getShape()[0])];
```

Comment:
I think `Math.toIntExact` would be better, since you're using it to construct an array.

    public static int toIntExact(long value)
    Returns the value of the long argument; throwing an exception if the value overflows an int.

https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#toIntExact-long-

    public static int saturatedCast(long value)
    Returns the same value cast to int if it is in the range of the int type, Integer.MAX_VALUE if it is too large, or Integer.MIN_VALUE if it is too small

https://google.github.io/guava/releases/19.0/api/docs/com/google/common/primitives/Ints.html

Reply:
Agreed - I've ended up going for `checkedCast` which is the same behaviour as `Math.toIntExact`

Reply:
The thing is, we are already using `toIntExact` elsewhere in the code base - I think it's important to maintain consistency. I don't have a strong opinion on which of the two we use.